packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -112,6 +116,80 @@ export async function convertWireToOsdkObjects(
   return ret;
 }
 
+export async function convertWireToOsdkObjects2(
+  client: MinimalClient,
+  objects: OntologyObjectV2[],
+  interfaceApiName: string | undefined,
+  forceRemoveRid: boolean = false,
+  selectedProps?: ReadonlyArray<string>,
+  strictNonNull: NullabilityAdherence = false,
+  interfaceToObjectTypeMappings: Record<
+    InterfaceTypeApiName,
+    InterfaceToObjectTypeMappings
+  > = {},
+): Promise<Osdk.Instance<ObjectOrInterfaceDefinition>[]> {
+  client.logger?.debug(`START convertWireToOsdkObjects2()`);
+
+  fixObjectPropertiesInPlace(objects, forceRemoveRid);
+
+  const ret = [];
+  for (const rawObj of objects) {
+    const objectDef = await client.ontologyProvider.getObjectDefinition(
+      rawObj.$apiName,
+    );
+    invariant(objectDef, `Missing definition for '${rawObj.$apiName}'`);
+
+    const interfaceToObjMapping = interfaceApiName
+      ? interfaceToObjectTypeMappings[interfaceApiName as InterfaceTypeApiName][
+        rawObj.$apiName
+      ]
+      : undefined;
+
+    const ifaceSelected = interfaceApiName && interfaceToObjMapping
+      ? (selectedProps
+        ? Object.keys(interfaceToObjMapping).filter(
+          val => {
+            selectedProps?.includes(interfaceToObjMapping[val]);
+          },
+        )
+        : [
+          ...Object.values(interfaceToObjMapping),
+          objectDef.primaryKeyApiName,
+        ])
+      : undefined;
+
+    // default value for when we are checking an object
+    let objProps;
+
+    let conforming = true;
+    if (interfaceApiName && ifaceSelected) {
+      invariantInterfacesAsViews(objectDef, interfaceApiName, client);
+
+      objProps = ifaceSelected;
```

Comment:
Again, we can just directly set this without converting because the props come over the wire as the object props, not the interface ones.